<!-- -->
<launch>

  <arg name="use_pacmod" default="true" />
  <arg name="kvaser_hardware_id" default="12345" />
  <arg name="kvaser_circuit_id" default="0" />
  <arg name="pacmod_vehicle_type" default="LEXUS_RX_450H" />

  <group ns="pacmod" if="$(arg use_pacmod)">
    <include file="$(find pacmod3)/launch/pacmod3.launch">
			<arg name="use_kvaser" value="true" />
      <arg name="pacmod_vehicle_type" value="$(arg pacmod_vehicle_type)" />
      <arg name="kvaser_hardware_id" value="$(arg kvaser_hardware_id)" />
      <arg name="kvaser_circuit_id" value="$(arg kvaser_circuit_id)" />
    </include>
  </group>

	<node pkg="as" type="pacmod_interface" name="pacmod_interface" output="screen">
    <rosparam command="load" file="$(find as)/config/pacmod_interface.yaml" />
    <param name="throttle_table_path" value="$(find as)/config/lexus_throttle.csv" />
    <param name="brake_table_path" value="$(find as)/config/lexus_brake.csv" />
    <remap from="pacmod/as_rx/steer_cmd" to= "psc/pacmod/as_rx/steer_cmd" />
    <remap from="pacmod/as_rx/accel_cmd" to= "psc/pacmod/as_rx/accel_cmd" />
    <remap from="pacmod/as_rx/brake_cmd" to= "psc/pacmod/as_rx/brake_cmd" />
    <remap from="pacmod/as_rx/shift_cmd" to= "psc/pacmod/as_rx/shift_cmd" />
    <remap from="pacmod/as_rx/turn_cmd" to= "psc/pacmod/as_rx/turn_cmd" />
	</node>

</launch>
